ansible_collections/arista/avd/roles/eos_designs/python_modules/interface_descriptions/__init__.py

```diff
@@ -0,0 +1,165 @@
+
+import importlib
+from collections import ChainMap
+from functools import cached_property
+from ansible_collections.arista.avd.plugins.plugin_utils.avdfacts import AvdFacts
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get, AristaAvdError
+
+
+DEFAULT_AVD_INTERFACE_DESCRIPTIONS_PYTHON_MODULE = (
+    "ansible_collections.arista.avd.roles.eos_designs.python_modules.interface_descriptions"
+)
+DEFAULT_AVD_INTERFACE_DESCRIPTIONS_PYTHON_CLASS_NAME = "AvdInterfaceDescriptions"
+
+
+class AvdInterfaceDescriptions(AvdFacts):
+
+    '''
+    Class used to render Interface Descriptions either from custom Jinja2 templates or using default Python Logic
+
+    Since some templates might contain certain legacy variables (switch_*),
+    those are mapped from the switch.* model
+
+    This class is imported adhoc based on the variable `templates.interface_descriptions.python_module` so it can
+    be overridden by a custom python class.
+    '''
+
+    @cached_property
+    def _default_mpls_overlay_role(self) -> str:
+        return get(self._hostvars, "switch.default_mpls_overlay_role")
+
+    @cached_property
+    def _mpls_lsr(self) -> str:
+        return get(self._hostvars, "switch.mpls_lsr")
+
+    @cached_property
+    def _mlag_peer(self) -> str:
+        return get(self._hostvars, "switch.mlag_peer", required=True)
+
+    @cached_property
+    def _mlag_port_channel_id(self) -> str:
```

Comment:
Given that we compute these cached_properties in these classes - would this mean that if the same was needed in another classes we would need to add the cached_property definition there as well?

Reply:
It is only locally significant for the instance of the class. Meaning the cache will be on instance level. Two instances of the class will _not_ share cache, but multiple requests for the same property on the same class will be returned from cache instead of running the method.